packages/docusaurus-utils-validation/src/validationSchemas.ts

```diff
@@ -8,7 +8,7 @@ import * as Joi from 'joi';
 import {isValidPathname} from '@docusaurus/utils';
 
 export const PluginIdSchema = Joi.string()
-  .regex(/^[a-zA-Z_\-]+$/)
+  .regex(/^[a-zA-Z_-]+$/)
```

Comment:
Removed useless escape. Covered by this test:

https://github.com/SamChou19815/docusaurus/blob/b0739a55da6dc8eedef7ecc172c55c1f88c474bd/packages/docusaurus-utils-validation/src/__tests__/validationSchemas.test.ts#L74